src/lib/octokit.js

```diff
@@ -2,22 +2,36 @@ const { Octokit } = require('@octokit/rest')
 const { throttling } = require('@octokit/plugin-throttling')
 const { retry } = require('@octokit/plugin-retry')
 const OctokitInstance = Octokit.plugin(throttling, retry)
+const logger = require('../lib/logger')
 
 let client
 
 module.exports = token => {
 	if (!client) {
 		client = new OctokitInstance({
 			previews: ['inertia-preview'],
-			auth: `token ${token}`,
+			auth: token,
 			throttle: {
-				onRateLimit: (retryAfter, options) => {
-					// Only retry once.
-					if (options.request.retryCount === 0) {
-						return true
-					}
+				onRateLimit: retryAfter => {
+					logger.log(
+						`Hit GitHub API rate limit, retrying after ${retryAfter}s`,
+						{
+							status: 'pending',
+							message: `Hit GitHub API rate limit, retrying after ${retryAfter}s`,
+						},
+					)
```

Comment:
**question:** do we not still want to set a retry count limit so we don't end up retrying rate limited calls forever?

Reply:
That's a really good point, I haven't thought of it because the [other project](https://github.com/BetaHuhn/repo-file-sync-action/issues/126) didn't put a retry count limit. perhaps we can put 5 as a magic number for retry count limit